lib/queue.js

```diff
@@ -505,7 +505,14 @@ Queue.prototype.processStalledJob = function(job){
           if(!isMember){
             _this.emit('stalled', job);
             return _this.processJob(job, true);
+          } else {
+            return job.releaseLock(_this.token);
           }
+        }).catch(function(err) {
+          // Any uncaught error will come here. We'll ensure that the job lock is released and rethrow
+          // the error so it bubbles normally
+          job.releaseLock(_this.token);
+          throw err;
         });
```

Comment:
I thought about using finally, but then I realized If we release an already released lock, we could  inadvertently release it for a resource that took the lock lawfully. With finally we are in a situation where we release the lock _at least once_, but with catch, we ensure that the lock is released _exactly once_ thus not running the risk of releasing the lock for some other consumer.